shopping_list_frontend/lib/cubits/user/registration_cubit.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:shopping_list_frontend/exceptions/users.dart';
+
+import '../../services/api/users.dart';
+
+abstract class RegistrationStatus {}
+
+class RegistrationStatusInitial extends RegistrationStatus {}
+
+class RegistrationStatusLoading extends RegistrationStatus {}
+
+class RegistrationStatusCheckFailed extends RegistrationStatus {
+  final String errorMessage;
+
+  RegistrationStatusCheckFailed(this.errorMessage);
+}
+
+class RegistrationStatusSuccessful extends RegistrationStatus {}
+
+abstract class TermsOfServiceStatus {}
+
+class TermsOfServiceRequiresConfirmation extends TermsOfServiceStatus {
+  static const int attemptsCount = 9;
+  int failedAttemptsCount;
+
+  TermsOfServiceRequiresConfirmation(this.failedAttemptsCount);
+}
+
+class TermsOfServiceAttemptLimitReached extends TermsOfServiceStatus {}
+
+class RegistrationState {
+  RegistrationStatus registrationStatus;
+  TermsOfServiceStatus termsOfServiceStatus;
+
+  RegistrationState({
+    required this.termsOfServiceStatus,
+    required this.registrationStatus,
+  });
+
+  RegistrationState copyWith({
+    RegistrationStatus? registrationStatus,
+    TermsOfServiceStatus? termsOfServiceStatus,
+  }) {
+    return RegistrationState(
+      registrationStatus: registrationStatus ?? this.registrationStatus,
+      termsOfServiceStatus: termsOfServiceStatus ?? this.termsOfServiceStatus,
+    );
+  }
+}
+
+class RegistrationCubit extends Cubit<RegistrationState> {
+  UserApi userApi = UserApi();
```

Comment:
please use provider for api, and make it as a final field
like this
```
class RegistrationCubit extends Cubit<RegistrationState> {
  final UserApi _userApi;
  RegistrationCubit(this._userApi)
      : super(
          RegistrationState(
            termsOfServiceStatus: TermsOfServiceRequiresConfirmation(0),
            registrationStatus: RegistrationStatusInitial(),
          ),
        );
....your code
}
```
And when you init object for this cubit you should do next
```
BlocProvider(
create:(context)=> RegistrationCubit(context.read<UserApi>()),
child: ....
)
```